File: ce/snippets/csharp/CRMV8/query/cs/fetchaggregationandgroupby5.cs
// <snippetfetchaggregationandgroupby5>


 // *****************************************************************************************************************
 //                FetchXML      estimatedvalue_max   Aggregate 5
 // *****************************************************************************************************************
 // Fetch the maximum estimatedvalue of all opportunities.  This is the equivalent of 
 // SELECT MAX(estimatedvalue) AS estimatedvalue_max ... in SQL.
 string estimatedvalue_max = @" 
 <fetch distinct='false' mapping='logical' aggregate='true'> 
     <entity name='opportunity'> 
        <attribute name='estimatedvalue' alias='estimatedvalue_max' aggregate='max' /> 
     </entity> 
 </fetch>";

 EntityCollection estimatedvalue_max_result = _serviceProxy.RetrieveMultiple(new FetchExpression(estimatedvalue_max));

 foreach (var c in estimatedvalue_max_result.Entities)
 {
     decimal aggregate5 = ((Money)((AliasedValue)c["estimatedvalue_max"]).Value).Value;
     System.Console.WriteLine("Max estimated value of all opportunities: " + aggregate5);

 }
 System.Console.WriteLine("===============================");

// </snippetfetchaggregationandgroupby5>

File: ce/snippets/visualbasic/CRMV8/queryvb/vb/fetchaggregationandgroupby5.vb
' <snippetfetchaggregationandgroupby5>


 ' *****************************************************************************************************************
 '                FetchXML      estimatedvalue_max   Aggregate 5
 ' *****************************************************************************************************************
 ' Fetch the maximum estimatedvalue of all opportunities.  This is the equivalent of 
 ' SELECT MAX(estimatedvalue) AS estimatedvalue_max ... in SQL.
 Dim estimatedvalue_max As String = " " &amp; ControlChars.CrLf &amp; _
     "                    <fetch distinct='false' mapping='logical' aggregate='true'> " &amp; ControlChars.CrLf &amp; _
     "                        <entity name='opportunity'> " &amp; ControlChars.CrLf &amp; _
     "                           <attribute name='estimatedvalue' alias='estimatedvalue_max' aggregate='max' /> " &amp; ControlChars.CrLf &amp; _
     "                        </entity> " &amp; ControlChars.CrLf &amp; _
     "                    </fetch>"

 Dim estimatedvalue_max_result As EntityCollection = _serviceProxy.RetrieveMultiple(New FetchExpression(estimatedvalue_max))

 For Each c In estimatedvalue_max_result.Entities
     Dim aggregate5 As Decimal = (CType((CType(c("estimatedvalue_max"), AliasedValue)).Value, Money)).Value
     Console.WriteLine("Max estimated value of all opportunities: " &amp; aggregate5)

 Next c
 Console.WriteLine("===============================")

' </snippetfetchaggregationandgroupby5>